server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/FilterToSearchQuery.java

```diff
@@ -0,0 +1,110 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.utils;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.apache.james.jmap.model.Filter;
+import org.apache.james.jmap.model.FilterCondition;
+import org.apache.james.jmap.model.FilterOperator;
+import org.apache.james.mailbox.model.SearchQuery;
+import org.apache.james.mailbox.model.SearchQuery.AddressType;
+import org.apache.james.mailbox.model.SearchQuery.Criterion;
+
+public class FilterToSearchQuery {
+
+    public SearchQuery map(Filter filter) {
+        if (filter instanceof FilterCondition) {
+            return mapCondition((FilterCondition) filter);
+        }
+        if (filter instanceof FilterOperator) {
+            return mapOperator((FilterOperator) filter);
+        }
+        throw new RuntimeException("Unknown filter");
+    }
+
+    private SearchQuery mapCondition(FilterCondition filter) {
+        SearchQuery searchQuery = new SearchQuery();
+        if (filter.getText().isPresent()) {
+            String text = filter.getText().get();
+            searchQuery.andCriteria(
```

Comment:
It's exactly what we said we should not do ... We have to index the concatenation of the fields but not search with a or, the result won't be correct anyway.